src/ui/digital-exchange/components/SearchBar.js

```diff
@@ -1,28 +1,50 @@
-import React from 'react';
+import React, { Component } from 'react';
 import PropTypes from 'prop-types';
 import { Button, Icon } from 'patternfly-react';
 import { Field, reduxForm } from 'redux-form';
 import SearchTextInput from 'ui/digital-exchange/components/SearchTextInput';
 
-const SearchBarBody = ({ clearSearch, handleSubmit }) => (
-  <form className="SearchBar__container" onSubmit={handleSubmit}>
-    <Field
-      name="keyword"
-      component={SearchTextInput}
-      onClear={clearSearch}
-    />
-    <Button
-      className="SearchBar__button"
-      type="submit"
-    >
-      <Icon name="search" />
-    </Button>
-  </form>
-);
+class SearchBarBody extends Component {
+  constructor(props) {
+    super(props);
+    this.clearSearch = this.clearSearch.bind(this);
+  }
+
+  clearSearch() {
+    const { reset, submit, searchTerm } = this.props;
+    reset();
+    if (searchTerm !== '') setTimeout(submit, 10);
```

Comment:
why is the timeout needed?

Reply:
hi @gidesan, this is because I also wanted to reset the results of the list thus removing the search keyword filter.

Reply:
is there any cleaner way to do this?

Reply:
if you mean by avoiding setTimeout, I could use onSubmit() - without parameters - to avoid waiting for reset and to achieve same result. would that be okay?